feature_engine/selection/drop_psi_features.py

```diff
@@ -0,0 +1,537 @@
+
+            - Considering all observations. In that case a split fraction of 0.25
+            will ensure 25% of the observations are in the "below cut-off"
+            dataframe. The final number of observations in the "below cut-off" may
+            be slightly different than 25% as all observations with the same value
+            are put in the same split.
+
+            - Considering the values of the observations. The cut-off applies to
+            the distinct values. In that case a split fraction of 0.25 will ensure
+            25% of the distinct values are in the "below cut-off" dataframe;
+            regardless of the number of observations.
+
+        Parameters
+        ----------
+        X : pandas dataframe
+            The original dataset. Correspond either to the measurement dataset
+            or will be spit into a measurement and a basis dataframe.
+
+        Returns
+        -------
+        pandas dataframe with value below the
+        basis pandas dataframe
+        """
+        # Identify the values according to which the split must be done.
+        if self.split_col == "use_df_index":
+            reference = pd.Series(X.index.to_list()).dropna()
+        else:
+            reference = X[self.split_col].dropna()
+
+        # Define the cut-off point based on quantile.
+        if self.split_distinct_value:
+            cut_off = self._get_cut_off_value(reference.unique())
+        else:
+            cut_off = self._get_cut_off_value(reference)
+
+        self.quantile = {self.split_col: cut_off}
+
+        # Split the original dataframe in two parts: above and below cut-off
+        is_above_cut_off = reference > cut_off
+
+        below_cut_off = X[~is_above_cut_off]
+        above_cut_off = X[is_above_cut_off]
+
+        return below_cut_off, above_cut_off
+
+    def _get_cut_off_value(self, ref):
+        """
+        Define the cut-off of the series (ref) in order to split the dataframe.
+
+        - For a float or integer, np.quantile is used.
+        - For other type, the quantile is based on the value_counts method.
+        This allows to deal with date or string in a unified way.
+            - The distinct values are sorted and the cumulative sum is
+            used to compute the quantile. The value with the quantile that
+            is the closest to the chosen split fraction is used as cut-off.
+            - The sort involves that categorical values are sorted alphabetically.
+
+        Parameters
+        ----------
+        ref : (np.array, pd.Series).
+            Series for which the nth quantile must be computed.
+
+        Returns
+        -------
+        cut_off: (float, int, str, object).
+            value for the cut-off.
+        """
+        # Ensure the argument is a pd.Series
+        if not isinstance(ref, pd.Series):
+            ref = pd.Series(ref)
+
+        # If the value is numerical, use numpy functionalities
+        if isinstance(ref.iloc[0], (int, float)):
```

Comment:
when would the user pass a numerical column to divide the dataframe? instead of dates I mean? just the index? or anything else?